src/container-type-context.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import {useScreenSize} from './hooks';
+
+import type {ContainerType} from './utils/types';
+
+const ContainerTypeContext = React.createContext<ContainerType | null>(null);
+
+export const useContainerType = (): ContainerType => {
+    const containerType = React.useContext(ContainerTypeContext);
+    const {isTablet, isDesktopOrBigger, isLargeDesktop} = useScreenSize();
```

Comment:
I won't mix screen size with container width, see my previous comment

Reply:
I've answered, I agree we should not mix, thats why I'm not mixing at all XD, I've created the Columns alike enum to just denote a different concept: not the same (conceptually) a mobile screen than a mobile column